lib/generators/rails/serializer_generator.rb

```diff
@@ -23,12 +23,14 @@ def association_names
       end
 
       def parent_class_name
-        if options[:parent]
-          options[:parent]
-        elsif defined?(::ApplicationSerializer)
-          'ApplicationSerializer'
+        if parent = options[:parent]
+          parent
         else
-          'ActiveModel::Serializer'
+          namespace = class_path.map(&:camelize).join('::')
+          application_serializer = [namespace, 'ApplicationSerializer']
+            .reject(&:blank?).join('::')
+          path = File.join('app/serializers', class_path, "application_serializer.rb")
```

Comment:
Yep, I gave that a shot based on your original comment. Does not seem to work, I believe because these files aren't `required` at this time.